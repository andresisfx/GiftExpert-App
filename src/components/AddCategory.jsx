import React from 'react'
import { useState } from 'react'
import PropTypes from 'prop-types'
export const AddCategory=({onNewValue})=> {

    const [inputValue, setInputValue] = useState('');
  const  handleInputChange=({target})=>{
         setInputValue(target.value)
    }
    const handleOnSubmit=(e)=>{
        e.preventDefault();
        const valueSpaced=inputValue.trim();
        if(valueSpaced.length<=2)return
        // de esta forma se recibe la funcion del estado y se la actualiza con el input value El callback recibe el valor más actualizado del estado anterior como argumento Cuando la nueva actualización depende directamente del estado anterior, como agregar un elemento a una lista.
        
       // setCategorias(categorias=>[inputValue,...categorias])

        onNewValue(valueSpaced.trim());
        setInputValue('')
    }
  return (
    <form onSubmit={handleOnSubmit} aria-label='form'>
      <input 
      type="text"
      placeholder='Buscar gifs'
      name='search'
      value={inputValue}
      onChange={handleInputChange}
       />
    </form>
  )
}

AddCategory.propTypes = {
  onNewValue:PropTypes.func.isRequired
}