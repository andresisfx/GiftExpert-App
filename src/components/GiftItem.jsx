import React from 'react'

export const  GiftItem=({title, url,id})=> {
    
  return (
    <div className="card">
        <h1>{title}</h1>
        <img src={url} alt={title} />
    </div>
  )
}

