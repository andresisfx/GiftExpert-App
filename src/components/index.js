export * from './GiftGrid.jsx'
export * from './GiftItem.jsx'
export * from './AddCategory.jsx'