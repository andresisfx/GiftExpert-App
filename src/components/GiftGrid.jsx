import React, {useState, useEffect } from 'react'
import {GiftItem} from './GiftItem';
import {useFetchGifts} from '../hooks/useFetchGifts'
import proptypes from 'prop-types';

export const GiftGrid=({category})=> {
  
  
const {images, isLoading} = useFetchGifts(category)

console.log("images GiftGrid",images)

console.log ("Here category",category)
      return (
        <>
            <h3>{category}</h3>

            {
                isLoading && ( <h2>Cargando...</h2> )
            }
        
            <div className="card-grid">
                  {
                      images.map((image)=> 
                      (
                        <GiftItem
                         key={image.id} 
                         {...image}/>
                      )
                      )
                  }
                
            </div>
        </>
      )
}

GiftGrid.proptypes={
    category:proptypes.string.isRequired
}