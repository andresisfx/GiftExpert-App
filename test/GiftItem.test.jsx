import { screen, render } from '@testing-library/react';
import { GiftItem } from '../src/components/GiftItem';

describe('Pruebas en GiftItem', () => { 
   
    const title='Hola Mundo'
    const url='https://localhost/algo.jpg'

    test('debe de hacer match con el snapshot', () => { 
        
        
        const {container}= render(<GiftItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
     });

    test( 'debe de mostrar la imagen con el url y el alt indicado', () => { 
        
        render (<GiftItem title={title} url={url} />)
        //screen.debug();
        // expect(screen.getByRole('img')).toBeTruthy();
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
       
     }); 

    test( 'debe de mostrar el titulo en el componente', () => { 
        
        render (<GiftItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy();
     });     
 })