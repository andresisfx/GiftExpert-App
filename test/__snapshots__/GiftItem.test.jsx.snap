// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftItem debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <h1>
      Hola Mundo
    </h1>
    <img
      alt="Hola Mundo"
      src="https://localhost/algo.jpg"
    />
  </div>
</div>
`;
